crates/net/network/src/transactions.rs

```diff
@@ -1074,116 +1255,287 @@ struct Peer {
 ///
 /// This will keep track of unique transaction hashes that are currently being fetched and submits
 /// new requests on announced hashes.
-#[derive(Debug, Default)]
+#[derive(Debug)]
+#[pin_project]
 struct TransactionFetcher {
+    /// All peers to which a request for pooled transactions is currently active. Maps 1-1 to
+    /// `inflight_requests`.
+    active_peers: LruMap<PeerId, u8, ByLength>,
     /// All currently active requests for pooled transactions.
+    #[pin]
     inflight_requests: FuturesUnordered<GetPooledTxRequestFut>,
-    /// Set that tracks all hashes that are currently being fetched.
-    inflight_hash_to_fallback_peers: HashMap<TxHash, Vec<PeerId>>,
+    /// Hashes that are awaiting fetch from an idle peer.
+    buffered_hashes: LruCache<TxHash>,
+    /// Tracks all hashes that are currently being fetched or are buffered, mapping them to
+    /// request retries and last recently seen fallback peers (max one request try for any peer).
+    unknown_hashes: LruMap<TxHash, (u8, LruCache<PeerId>), Unlimited>,
+    /// Size metadata for unknown eth68 hashes.
+    eth68_meta: LruMap<TxHash, usize, Unlimited>,
```

Comment:
are we using any LRU features for this map?
if not then this is just a HashMap?